hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreFlusher.java

```diff
@@ -495,41 +505,54 @@ public int getFlushQueueSize() {
    * Only interrupt once it's done with a run through the work loop.
    */
   void interruptIfNecessary() {
-    lock.writeLock().lock();
+    lock.readLock().lock();
```

Comment:
Mind explaining a bit why readLock is enough here?

Reply:
Thanks for reviewing Duo. 

After digging and reconsideration, I think we should keep the writeLock here. Sorry for the mistake.